serveradmin/serverdb/views.py

```diff
@@ -87,38 +100,35 @@ def history(request):
     if not object_id:
         raise Http404
 
-    adds = ChangeAdd.objects.filter(server_id=object_id)
-    updates = ChangeUpdate.objects.filter(server_id=object_id)
-    deletes = ChangeDelete.objects.filter(server_id=object_id).select_related()
+    # TODO: Change data model so that server_id is moved to ChangeCommit.
+    #       To me this would make most sense. If that is not possible (why?!)
+    #       write a proper (custom) query that performs.
+    where = {'server_id': object_id}
+    adds = ChangeAdd.objects.filter(**where)
+    updates = ChangeUpdate.objects.filter(**where)
+    deletes = ChangeDelete.objects.filter(**where)
 
     # @TODO Transform to PostgreSQL JSON and put a index here
     if search_string:
         adds = adds.filter(attributes_json__contains=search_string)
         updates = updates.filter(updates_json__contains=search_string)
         deletes = deletes.filter(attributes_json__contains=search_string)
 
-    adds = adds.select_related()
-    updates = updates.select_related()
-
     if commit_id:
         adds = adds.filter(commit__pk=commit_id)
         updates = updates.filter(commit__pk=commit_id)
         deletes = deletes.filter(commit__pk=commit_id)
 
+    related = ('commit__app', 'commit__user')
```

Comment:
Ahh I wasn't aware that prefetching relations does not cascade to other relations - makes perfect sense.

Good catch! :+1: